src/main/java/com/gooddata/connector/ConnectorService.java

```diff
@@ -224,4 +225,42 @@ public void handlePollException(final GoodDataRestException e) {
     }
 
 
+    /**
+     * Polls on provided connector process
+     *
+     * @param process process to be executed
+     * @return executed process
+     * @throws ConnectorException if process execution fails
+     */
+    public FutureResult<ProcessStatus> pollProcess(final IntegrationProcessStatus process) {
+        notNull(process, "process");
+        notNull(process.getUri(), "process.getUri");
+
+        try {
+            return new PollResult<>(this, new SimplePollHandler<ProcessStatus>(process.getUri(), ProcessStatus.class) {
```

Comment:
Could you please reuse this PollResult also in executeProcess? It seems to be the same one.